pkgs/by-name/ll/llama-cpp/package.nix

```diff
@@ -140,7 +144,7 @@ stdenv.mkDerivation (finalAttrs: {
     license = licenses.mit;
     mainProgram = "llama-cpp-main";
     maintainers = with maintainers; [ dit7ya elohmeier ];
-    broken = stdenv.isDarwin && stdenv.isx86_64;
+    broken = (effectiveStdenv.isDarwin && effectiveStdenv.isx86_64) || lib.count lib.id [openclSupport openblasSupport rocmSupport cudaSupport] == 0;
```

Comment:
> x86_64-darwin

Wait is that right? At least the build works fine in the upstream flake, not sure about the runtime

Reply:
trying to compile on x86_64 darwin I get the following failure

/tmp/nix-build-llama-cpp-1710.drv-0/source/ggml-metal.m:8:9: fatal error: 'M…
┃        > #import <Metal/Metal.h>

Reply:
Maybe something from https://github.com/ggerganov/llama.cpp/blob/e39106c0554cbd0e9310e08fb3b2a577ea4b6273/.devops/nix/package.nix#L83-L90 could help

Reply:
But also this `meta.broken` wasn't introduced by your PR and you don't necessarily have to fix it here. If you choose not to, would be nice to open an issue

Reply:
all the dependencies for darwin are there, maybe it could be a flag that is missing, not sure. I think it's better to leave this for a subsequent PR as I don't see the obvious fix.